pilot/pkg/bootstrap/istio_ca.go

```diff
@@ -479,6 +477,39 @@ func (s *Server) createIstioCA(opts *caOptions) (*ca.IstioCA, error) {
 	return istioCA, nil
 }
 
+func (s *Server) createSelfSignedCACertificateOptions(fileBundle *ca.SigningCAFileBundle, opts *caOptions) (*ca.IstioCAOptions, error) {
+	var caOpts *ca.IstioCAOptions
+	var err error
+	if s.kubeClient != nil {
+		log.Info("Use self-signed certificate as the CA certificate")
+
+		// Abort after 20 minutes.
```

Comment:
How did we come up with 20 minutes?

Reply:
Not sure where 20 min originally came from.

Reply:
It is the retry timeout for getting secret from k8s